### PR TITLE
Implement Logging

### PR DESCRIPTION
**Pull request by Aryan-sharma11**
This PR is to track changes regarding Logging https://github.com/biosharp-dotnet/sucuri-api/issues/23

### CODE REVIEW

**Comment by abhi18av** [comment]
@Aryan-sharma11 , could you please add some console outputs/ screenshots with the PRs. This facilitates the review process.

File: README.md [hunk 1]
```diff
@@ -1,15 +1,52 @@
...
+
+ ## Architecture
+```mermaid
+graph TB
+   subgraph "Client Side"
+   a[String with conda package name and conda channels ] --> b[Create a query]
+   f[Latest avaliable version]
+   end
+   subgraph "Server Side"
+   b --Request with Restish or Invoke-RestMethod--> c[Recieve string parameter]
```
Comment (abhi18av) [inline README.md:47]:
```suggestion
   b --Request--> c[Recieve string parameter]

```

Comment (abhi18av) [reply README.md:47]: A request can be triggered from any client.

File: README.md [hunk 2]
```diff
@@ -1,15 +1,52 @@
...
-   b --Request with Restish--> c[Recieve string parameter]
-   c --> d[Use micromamba binary]
-   d --> e[Format json output]
-   e --Return json string--> f
-   end
-```
+# SUCURI-API
+
+## Introduction 
+Sucuri -API is an api developed using Pode framework which uses conda binaries to provide latest version of any given conda package.
```
Comment (abhi18av) [inline README.md:4]:
```suggestion
Sucuri-API is an API developed using [Pode](https://badgerati.github.io/Pode) framework which uses conda binary (conda/mamba/micromamba) to provide latest version of any given conda package.

```

**Comment by Aryan-sharma11** [comment]
> @Aryan-sharma11 , could you please add some console outputs/ screenshots with the PRs. This facilitates the review process.

Hi , right now there's no additions , I have just created this PR to track changes, when I will make changes I will surely provide screenshots .🙂

**Comment by Aryan-sharma11** [comment]
@abhi18av  I have added logging for `errors` and `request`, right now we are storing the logs in a folder `/logs` which will be stored for maximum 1 day , we can increase it if needed  also. 
image

error logs
image

request logs
image

**Comment by Aryan-sharma11** [comment]
> Reference

Also I have doubt, that can we store `pkg_name ` in the logs ?

**Comment by abhi18av** [comment]
@Aryan-sharma11 , this is looking promising! 💯 

> Also I have doubt, that can we store pkg_name in the logs ?

Yes, because by default the information is stored only regarding the timing/user-agent which isn't very useful for us.

```
127.0.0.1 - - [25/Aug/2022:08:24:34 +02:00] "GET /v1/search HTTP/1.1" 200 47 "http://localhost:8085/docs/view" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36"
127.0.0.1 - - [25/Aug/2022:08:25:20 +02:00] "GET /v1/search HTTP/1.1" 200 47 "-" "curl/7.83.1"
```

**Comment by abhi18av** [comment]
@Aryan-sharma11 , I noticed that the `channels` are missing from the docs page - could you please add those in the openapi spec

**Comment by Aryan-sharma11** [comment]
> Yes, because by default the information is stored only regarding the timing/user-agent which isn't very useful for us.
@abhi18av  I need help regarding how to do it in `Pode`, I think we have to build some custom logic but I am not able to understand  how to do it 😥

**Comment by abhi18av** [comment]
@Aryan-sharma11 , no problem - I think that we can now request @Mxrcon to take a look into this one.

If not, then we can discuss this during the week via PRs and I'll allocate some time to this.

**Comment by abhi18av** [comment]
Task linked: [CU-2zard5d Finalize the logging logic](https://app.clickup.com/t/2zard5d)

**Pull request merged**